Make umount respect the -n option when using -r

Patch by Derick Swanepoel

http://bugs.gentoo.org/98675

--- mount/umount.c
+++ mount/umount.c
@@ -349,7 +349,8 @@
 			remnt.mnt_type = remnt.mnt_fsname = NULL;
 			remnt.mnt_dir = xstrdup(node);
 			remnt.mnt_opts = xstrdup("ro");
-			update_mtab(node, &remnt);
+			if (!nomtab && mtab_is_writable())
+				update_mtab(node, &remnt);
 			return 0;
 		} else if (errno != EBUSY) { 	/* hmm ... */
 			perror("remount");
